java-library/src/main/java/com/microsoft/azure/functions/dapr/annotation/DaprBindingOutput.java

```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.functions.dapr.annotation;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+import com.microsoft.azure.functions.annotation.CustomBinding;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.ElementType;
+
+/**
+ * Attribute to specify parameters for the Dapr output bindings.
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.PARAMETER)
+@CustomBinding(direction = "out", name = "", type = "daprBinding")
+public @interface DaprBindingOutput {
+    /**
+     * The variable name used in function.json.
```

Comment:
There is no functions.json for Java, right?
```suggestion
     * Name of the dapr binding
```

Reply:
Looks like there is another param for binding name, what does this do? I notice that this "name" is in all annotations.

Reply:
name property is used for binding the name of the parameter that will carry the request data. It is used in language worker runtime to send and receive the data.

Reply:
Looking at 
<img width="799" alt="image" src="https://github.com/Azure/azure-functions-dapr-extension/assets/22132944/86834c62-2368-463f-87c2-f2325a1a49c3">


Binding output takes a parameter "name=state", but the actual binding is tied to `OutputBinding<string> product`, discarding `state` altogether. Also for end users, they don't really know what is function.json (it's an implementation detail for Java, since users don't manually write one). We should explore how to get around this.

Reply:
See this for example, does not have a `name` parameter https://github.com/Azure/azure-functions-rabbitmq-extension/blob/dev/java-library/src/main/java/com/microsoft/azure/functions/rabbitmq/annotation/RabbitMQOutput.java

Reply:
Yeah it seems, it is not required, but for some reason they have added name property in all the extension in [main java-library](https://github.com/Azure/azure-functions-java-library/blob/2387b6b1d8d114c919377a6ac66f9fe63f4d31cd/src/main/java/com/microsoft/azure/functions/annotation/BlobInput.java#L44-L49).

Reply:
We can check with the Java library folks, and before that maybe we can try removing the property and checking if it works (like RabbitMQ).

Reply:
I think [this is the reason RabbitMQ](https://github.com/Azure/azure-functions-rabbitmq-extension/blob/ec0bb66288027b8b095664cabdbd7d6a1ab16ffa/java-library/src/main/java/com/microsoft/azure/functions/rabbitmq/annotation/RabbitMQOutput.java#L41C35-L41C57) doesn't have to add name property to the annotation class, I made [similar change](https://github.com/Azure/azure-functions-dapr-extension/pull/131/files#diff-7ec424919c2f21b7b84506f5823ca389f503b738e500122248ef95a86d770b93)
for Dapr java-library.